The gem fails to compile from source on Fedora 35

This is on CRuby 3.0.3. I'm trying from source since I saw a very similar error on TruffleRuby.
So it seems an issue in this gem since it also does not work on CRuby.
It could also be due to my environment but it's a pretty much vanilla Fedora 35.
```
$ ruby -v
ruby 3.0.3p157 (2021-11-24 revision 3fb7d2cadc) [x86_64-linux]
```

```
$ gem i --platform=ruby -V sqlite3
HEAD https://index.rubygems.org/
200 OK
GET https://index.rubygems.org/info/sqlite3
200 OK
GET https://index.rubygems.org/info/ffi
200 OK
GET https://index.rubygems.org/info/mini_portile2
200 OK
Downloading gem sqlite3-1.5.2.gem
Downloading gem mini_portile2-2.8.0.gem
GET https://index.rubygems.org/gems/sqlite3-1.5.2.gem
Fetching sqlite3-1.5.2.gem
200 OK
GET https://index.rubygems.org/gems/mini_portile2-2.8.0.gem
Fetching mini_portile2-2.8.0.gem
200 OK
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/.github/FUNDING.yml
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/.github/workflows/ci.yml
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/.gitignore
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/CHANGELOG.md
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/Gemfile
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/LICENSE.txt
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/README.md
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/Rakefile
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/SECURITY.md
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/lib/mini_portile2.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/lib/mini_portile2/mini_portile.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/lib/mini_portile2/mini_portile_cmake.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/lib/mini_portile2/version.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/mini_portile2.gemspec
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/test/assets/git/config
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/test/assets/gpg-fixtures/data
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/test/assets/gpg-fixtures/data.asc
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/test/assets/gpg-fixtures/data.invalid.asc
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/test/assets/patch 1.diff
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/test/assets/test mini portile-1.0.0/configure
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/test/assets/test-cmake-1.0/CMakeLists.txt
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/test/assets/test-cmake-1.0/hello.c
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/test/assets/test-download-archive.tar.gz
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/test/helper.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/test/test_cmake.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/test/test_cook.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/test/test_digest.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/test/test_download.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/test/test_execute.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/mini_portile2-2.8.0/test/test_proxy.rb
Successfully installed mini_portile2-2.8.0
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/.gemtest
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/API_CHANGES.md
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/CHANGELOG.md
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/CONTRIBUTING.md
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ChangeLog.cvs
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/Gemfile
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/LICENSE
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/LICENSE-DEPENDENCIES
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/README.md
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/dependencies.yml
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ext/sqlite3/aggregator.c
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ext/sqlite3/aggregator.h
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ext/sqlite3/backup.c
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ext/sqlite3/backup.h
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ext/sqlite3/database.c
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ext/sqlite3/database.h
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ext/sqlite3/exception.c
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ext/sqlite3/exception.h
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ext/sqlite3/extconf.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ext/sqlite3/sqlite3.c
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ext/sqlite3/sqlite3_ruby.h
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ext/sqlite3/statement.c
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ext/sqlite3/statement.h
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/faq/faq.md
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/faq/faq.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/faq/faq.yml
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/lib/sqlite3.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/lib/sqlite3/constants.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/lib/sqlite3/database.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/lib/sqlite3/errors.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/lib/sqlite3/pragmas.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/lib/sqlite3/resultset.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/lib/sqlite3/statement.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/lib/sqlite3/translator.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/lib/sqlite3/value.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/lib/sqlite3/version.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ports/archives/sqlite-autoconf-3390400.tar.gz
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/test/helper.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/test/test_backup.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/test/test_collation.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/test/test_database.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/test/test_database_flags.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/test/test_database_readonly.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/test/test_database_readwrite.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/test/test_deprecated.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/test/test_encoding.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/test/test_integration.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/test/test_integration_aggregate.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/test/test_integration_open_close.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/test/test_integration_pending.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/test/test_integration_resultset.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/test/test_integration_statement.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/test/test_result_set.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/test/test_sqlite3.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/test/test_statement.rb
/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/test/test_statement_execute.rb
Building native extensions. This could take a while...
current directory: /home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ext/sqlite3
["/home/eregon/.rubies/ruby-3.0.3/bin/ruby", "-I", "/home/eregon/.rubies/ruby-3.0.3/lib/ruby/3.0.0", "-r", "./siteconf20221004-37471-9wjoba.rb", "extconf.rb"]
Building sqlite3-ruby using packaged sqlite3.
Extracting sqlite-autoconf-3390400.tar.gz into tmp/x86_64-redhat-linux/ports/sqlite3/3.39.4... OK
Running 'configure' for sqlite3 3.39.4... OK
Running 'compile' for sqlite3 3.39.4... OK
Running 'install' for sqlite3 3.39.4... OK
Activating sqlite3 3.39.4 (from /home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ports/x86_64-redhat-linux/sqlite3/3.39.4)...
checking for whether -lsqlite3 is accepted as LDFLAGS... yes
checking for whether -lz is accepted as LDFLAGS... yes
checking for whether -lm is accepted as LDFLAGS... yes
checking for sqlite3.h... yes
checking for sqlite3_libversion_number() in -lsqlite3... yes
checking for rb_proc_arity()... yes
checking for rb_integer_pack()... yes
checking for sqlite3_initialize()... yes
checking for sqlite3_backup_init()... yes
checking for sqlite3_column_database_name()... no
checking for sqlite3_enable_load_extension()... yes
checking for sqlite3_load_extension()... yes
checking for sqlite3_open_v2()... yes
checking for sqlite3_prepare_v2()... yes
checking for sqlite3_int64 in sqlite3.h... yes
checking for sqlite3_uint64 in sqlite3.h... yes
creating Makefile
current directory: /home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ext/sqlite3
["make", "DESTDIR=", "clean"]

current directory: /home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ext/sqlite3
["make", "DESTDIR="]
compiling aggregator.c
compiling backup.c
compiling database.c
compiling exception.c
compiling sqlite3.c
compiling statement.c
linking shared-object sqlite3/sqlite3_native.so
/usr/bin/ld: cannot find -lsqlite3
collect2: error: ld returned 1 exit status
make: *** [Makefile:262: sqlite3_native.so] Error 1
ERROR:  Error installing sqlite3:
	ERROR: Failed to build gem native extension.

    current directory: /home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ext/sqlite3
/home/eregon/.rubies/ruby-3.0.3/bin/ruby -I /home/eregon/.rubies/ruby-3.0.3/lib/ruby/3.0.0 -r ./siteconf20221004-37471-9wjoba.rb extconf.rb
current directory: /home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ext/sqlite3
make DESTDIR\= clean
current directory: /home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ext/sqlite3
make DESTDIR\=
Building has failed. See above output for more information on the failure.
make failed, exit code 2

Gem files will remain installed in /home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2 for inspection.
Results logged to /home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/extensions/x86_64-linux/3.0.0/sqlite3-1.5.2/gem_make.out
zsh: exit 1     gem i --platform=ruby -V sqlite3
```

```
$ cd /home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2
$ find . -name 'libsqlite*'       
./ext/sqlite3/tmp/x86_64-redhat-linux/ports/sqlite3/3.39.4/sqlite-autoconf-3390400/.libs/libsqlite3.a
./ext/sqlite3/tmp/x86_64-redhat-linux/ports/sqlite3/3.39.4/sqlite-autoconf-3390400/.libs/libsqlite3.lai
./ext/sqlite3/tmp/x86_64-redhat-linux/ports/sqlite3/3.39.4/sqlite-autoconf-3390400/.libs/libsqlite3.la
./ext/sqlite3/tmp/x86_64-redhat-linux/ports/sqlite3/3.39.4/sqlite-autoconf-3390400/libsqlite3.la
./ports/x86_64-redhat-linux/sqlite3/3.39.4/lib/libsqlite3.la
./ports/x86_64-redhat-linux/sqlite3/3.39.4/lib/libsqlite3.a
```

```
$ which pkg-config
/usr/bin/pkg-config
$ pkg-config --version
1.8.0
$ pkg-config --libs --static ./ports/x86_64-redhat-linux/sqlite3/3.39.4/lib/pkgconfig/sqlite3.pc
-L/home/eregon/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ports/x86_64-redhat-linux/sqlite3/3.39.4/lib -lsqlite3 -lz -lm 
```

```
In ~/.rubies/ruby-3.0.3/lib/ruby/gems/3.0.0/gems/sqlite3-1.5.2/ext/sqlite3 $ make V=1
rm -f sqlite3_native.so
gcc -shared -o sqlite3_native.so aggregator.o backup.o database.o exception.o sqlite3.o statement.o -L. -L/home/eregon/.rubies/ruby-3.0.3/lib -Wl,-rpath,/home/eregon/.rubies/ruby-3.0.3/lib -L. -L/home/eregon/.rubies/ruby-3.0.3/lib  -fstack-protector-strong -rdynamic -Wl,-export-dynamic  -lsqlite3 -lz -lm -L/home/eregon/.rubies/ruby-3.0.3/lib  -Wl,--compress-debug-sections=zlib    -Wl,-rpath,/home/eregon/.rubies/ruby-3.0.3/lib -L/home/eregon/.rubies/ruby-3.0.3/lib -lruby -lsqlite3  -lsqlite3 -lm   -lc
/usr/bin/ld: cannot find -lsqlite3
collect2: error: ld returned 1 exit status
make: *** [Makefile:262: sqlite3_native.so] Error 1
zsh: exit 2     make V=1
```
So the `-L` seems gone and then of course it cannot find libsqlite3.

--- a/.github/workflows/sqlite3-ruby.yml
+++ b/.github/workflows/sqlite3-ruby.yml
@@ -76,6 +76,21 @@ jobs:
       - run: bundle exec rake compile -- --enable-system-libraries
       - run: bundle exec rake test
 
+  # reported at https://github.com/sparklemotion/sqlite3-ruby/issues/354
+  # TODO remove once https://github.com/flavorjones/mini_portile/issues/118 is fixed
+  fedora:
+    runs-on: ubuntu-latest
+    container:
+      image: fedora:35
+    steps:
+      - run: |
+          dnf group install -y "C Development Tools and Libraries"
+          dnf install -y ruby ruby-devel
+      - uses: actions/checkout@v3
+      - run: bundle install
+      - run: bundle exec rake compile -- --disable-system-libraries
+      - run: bundle exec rake test
+
   sqlcipher:
     strategy:
       fail-fast: false

--- a/ext/sqlite3/extconf.rb
+++ b/ext/sqlite3/extconf.rb
@@ -66,15 +66,24 @@ module Sqlite3
           end
           recipe.activate
 
-          ENV["PKG_CONFIG_ALLOW_SYSTEM_CFLAGS"] = "t" # on macos, pkg-config will not return --cflags without this
-          pcfile = File.join(recipe.path, "lib", "pkgconfig", "sqlite3.pc")
-          if pkg_config(pcfile)
-            # see https://bugs.ruby-lang.org/issues/18490
-            libs = xpopen(["pkg-config", "--libs", "--static", pcfile], err: [:child, :out], &:read)
-            libs.split.each { |lib| append_ldflags(lib) } if $?.success?
-          else
-            abort("\nCould not configure the build properly. Please install either the `pkg-config` utility or the `pkg-config` rubygem.\n\n")
+          # on macos, pkg-config will not return --cflags without this
+          ENV["PKG_CONFIG_ALLOW_SYSTEM_CFLAGS"] = "t"
+
+          lib_path = File.join(recipe.path, "lib")
+          pcfile = File.join(lib_path, "pkgconfig", "sqlite3.pc")
+          abort_pkg_config("pkg_config") unless pkg_config(pcfile)
+
+          # see https://bugs.ruby-lang.org/issues/18490
+          flags = xpopen(["pkg-config", "--libs", "--static", pcfile], err: [:child, :out], &:read)
+          abort_pkg_config("xpopen") unless $?.success?
+          flags = flags.split
+
+          # see https://github.com/flavorjones/mini_portile/issues/118
+          "-L#{lib_path}".tap do |lib_path_flag|
+            flags.prepend(lib_path_flag) unless flags.include?(lib_path_flag)
           end
+
+          flags.each { |flag| append_ldflags(flag) }
         end
       end
 
@@ -138,6 +147,10 @@ module Sqlite3
 
       def abort_could_not_find(missing)
         abort("\nCould not find #{missing}.\nPlease visit https://github.com/sparklemotion/sqlite3-ruby for installation instructions.\n\n")
+      end
+
+      def abort_pkg_config(id)
+        abort("\nCould not configure the build properly (#{id}). Please install either the `pkg-config` utility or the `pkg-config` rubygem.\n\n")
       end
 
       def cross_build?

